### PR TITLE
chore(prettier): Adds prettier

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
Formatting gets messy with a team of 30 contributors to a repository. Time is wasted on simple common code formatting rules. And new developers first contribution can be painful because of that.

**What is the chosen solution to this problem?**
I'd like to start the discussion of adding [prettier](https://github.com/prettier/prettier) to Talend/ui.
Prettier is an opinionated but configurable code formater. It removes you contributor the hassle of conforming to the common formatting rules (Appart from a couple of extras from eslint ;)) and the reviewers from having to check for proper formatting on every reviews.

We've been using it without issues on our project for a couple of weeks with great results.
I'd like to have your feedback on whether or not we should add it. 

For webstorm : https://github.com/prettier/prettier/blob/master/editors/webstorm/README.md

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -3,10 +3,16 @@
   "parser": "babel-eslint",
   "rules": {
     "import/no-extraneous-dependencies": ["error", {"devDependencies": ['*.test.js', '.spec.js']}],
-    "indent": ["error", "tab"],
+		"indent": ["error", "tab", { "SwitchCase": 1 }],
     "react/prefer-es6-class": 0,
     "react/jsx-indent": ["error", "tab"],
     "react/jsx-indent-props": ["error", "tab"]
+		"new-cap": [
```
Comment (talend-sromain) [inline .eslintrc]: maybe some comma missing before this line ?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+	"printWidth": 100,
+	"tabWidth": 4,
```
Comment (talend-sromain) [inline .prettierrc]: is that really matters as we use tabs ?

File: output/cmf.eslint.txt [hunk 3]
```diff
@@ -2,19 +2,21 @@
...
-  35:2  warning  Unexpected constant condition  no-constant-condition
-
-✖ 4 problems (1 error, 3 warnings)
-
+Cannot read config file: /home/travis/build/Talend/ui/.eslintrc
+Error: missed comma between flow collection entries at line 10, column 3:
+    		"new-cap": [
+      ^
+YAMLException: Cannot read config file: /home/travis/build/Talend/ui/.eslintrc
+Error: missed comma between flow collection entries at line 10, column 3:
```
Comment (talend-sromain) [inline output/cmf.eslint.txt]: the missing comma has some consequences

**Comment by talend-sromain** [comment]
but i approve the idea 👍

File: output/forms.eslint.txt [hunk 4]
```diff
@@ -2,63 +2,21 @@
 > react-talend-forms@0.111.0 lint:es /home/travis/build/Talend/ui/packages/forms
 > eslint --config .eslintrc src
 
-The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.
-
-/home/travis/build/Talend/ui/packages/forms/src/fields/ArrayField.js
```
Comment (jare-talend) [inline output/forms.eslint.txt]: ignore some files to avoid to get diff with the original files (react-jsonschema-format)

**Pull request merged**